redpanda/cloud/controlplane.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package cloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"buf.build/gen/go/redpandadata/cloud/grpc/go/redpanda/api/controlplane/v1beta2/controlplanev1beta2grpc"
+	controlplanev1beta2 "buf.build/gen/go/redpandadata/cloud/protocolbuffers/go/redpanda/api/controlplane/v1beta2"
+	"google.golang.org/grpc"
+)
+
+// ControlPlaneClientSet holds the respective service clients to interact with
+// the control plane endpoints of the Public API.
+type ControlPlaneClientSet struct {
```

Comment:
hm, this is strange, we shouldn't be using the control plane API from the Terraform provider. What am I missing? It has to be coded against the Public Cloud API.

Reply:
If it is the actual cloud API, then it seems the package is wrongly named.

Reply:
We are not, we are using the control plane section of the public api in this struct (network, resource group and cluster) hence the comment

```
// ControlPlaneClientSet holds the respective service clients to interact with
// the control plane endpoints of the Public API.
```

Reply:
> If it is the actual cloud API, then it seems the package is wrongly named.

could you provide a suggestion for the package name please? 

It is the package that creates the clients to speak with Redpanda cloud.